utils/build_swift/driver_arguments.py

```diff
@@ -0,0 +1,1295 @@
+           help='user-visible version of the embedded Clang and LLVM '
+                'compilers')
+
+    option('--swift-compiler-version', set_,
+           metavar='VERSION',
+           help='Swift compiler version')
+
+    option('--swift-user-visible-version', set_,
+           default=defaults.SWIFT_USER_VISIBLE_VERSION,
+           metavar='VERSION',
+           help='user-visible version of the embedded Swift compiler')
+
+    # -------------------------------------------------------------------------
+    in_group('CMake')
+
+    # FIXME: This should be one option using choices
+    with mutually_exclusive_group():
+        set_defaults(cmake_generator=defaults.CMAKE_GENERATOR)
+
+        # FIXME: '-n' is taken by dry-run
+        option(['-N', '--ninja'],
+               set_('cmake_generator', const='Ninja'),
+               help='use Cmake\'s Ninja generator (default)')
+
+        option(['-x', '--xcode'],
+               set_('cmake_generator', const='Xcode'),
+               help='use CMake\'s Xcode generator (default is Ninja)')
+
+        option(['-m', '--make'],
+               set_('cmake_generator', const='Unix Makefiles'),
+               help='use CMake\'s Makefile generator (default is Ninja)')
+
+        option(['-e', '--elipse'],
+               set_('cmake_generator', const='Eclipse CDT4 - Ninja'),
+               help='use CMake\'s Makefile generator (default is Ninja)')
+
+        option('--cmake-generator', set_,
+               default='Ninja',
+               help='manually select the CMake generator')
+
+    # TODO: Remove the 'extra' variant in favor of simpler '--cmake-options'
+    # TODO: Add '-D' option to directly pass CMake arguments
+    option('--extra-cmake-options',
+           append('cmake_options', separator=','),
+           help='pass through extra options to CMake in the form of comma '
+                'separated options "-DCMAKE_VAR1=YES,-DCMAKE_VAR2=/tmp". '
+                'Can be called multiple times to add multiple such options.')
+
+    # -------------------------------------------------------------------------
+    in_group('Build Variants')
+
+    with mutually_exclusive_group():
+        all_build_variants = [
+            'cmark_build_variant',
+            'foundation_build_variant',
+            'llbuild_build_variant',
+            'lldb_build_variant',
+            'libdispatch_build_variant',
+            'libicu_build_variant',
+            'llvm_build_variant',
+            'swift_build_variant',
+            'swift_stdlib_build_variant',
+        ]
+
+        set_defaults(all_build_variants, defaults.BUILD_VARIANT)
+
+        option(['-d', '--debug'],
+               set_(all_build_variants, const='Debug'),
+               help='build the Debug variant of everything (LLVM, Clang, '
+                    'Swift host tools, target Swift standard libraries, LLDB '
+                    '(if enabled) (default)')
+
+        option(['-r', '--release-debug'],
```

Comment:
Can you keep this as --release-debuginfo. Otherwise, it is ambiguous  against --debug.